test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -108,111 +104,117 @@ public async Task StandbyMode_EndToEnd_LinuxContainer()
-                Assert.Equal(1, logLines.Count(p => p.Contains($"Starting Host (HostId={_expectedHostId}")));
-                Assert.Contains("Node.js HttpTrigger function invoked.", logLines);
-
-                // verify cold start log entry
-                var coldStartLog = _loggerProvider.GetAllLogMessages().FirstOrDefault(p => p.Category == ScriptConstants.LogCategoryHostMetrics);
-                JObject coldStartData = JObject.Parse(coldStartLog.FormattedMessage);
-                Assert.Equal("Dynamic", coldStartData["sku"]);
-                Assert.True((int)coldStartData["dispatchDuration"] > 0);
-                Assert.True((int)coldStartData["functionDuration"] > 0);
-            }
+
+            var environment = new TestEnvironment(vars);
+
+            await InitializeTestHostAsync("Linux", environment);
+
+            await VerifyWarmupSucceeds();
+            await VerifyWarmupSucceeds(restart: true);
+
+            // now specialize the site
+            await Assign(encryptionKey);
+
+            // immediately call a function - expect the call to block until
+            // the host is fully specialized
+            // the Unauthorized is expected since we havne't specified the key
+            // it's enough here to ensure we don't get a 404
+            var request = new HttpRequestMessage(HttpMethod.Get, $"api/httptrigger");
+            request.Headers.Add(ScriptConstants.AntaresColdStartHeaderName, "1");
+            var response = await _httpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+
+            // now that the host is initialized, send a valid key
+            // and expect success
+            var secretManager = _httpServer.Host.Services.GetService<ISecretManagerProvider>().Current;
+            var keys = await secretManager.GetFunctionSecretsAsync("HttpTrigger");
+            string key = keys.First().Value;
+            request = new HttpRequestMessage(HttpMethod.Get, $"api/httptrigger?code={key}");
+            request.Headers.Add(ScriptConstants.AntaresColdStartHeaderName, "1");
+            response = await _httpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Assert.False(environment.IsPlaceholderModeEnabled());
+            Assert.True(environment.IsContainerReady());
+
+            // verify warmup function no longer there
+            request = new HttpRequestMessage(HttpMethod.Get, "api/warmup");
+            response = await _httpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+
+            _httpServer.Dispose();
+            _httpClient.Dispose();
+
+            string sanitizedMachineName = Environment.MachineName
+                    .Where(char.IsLetterOrDigit)
+                    .Aggregate(new StringBuilder(), (b, c) => b.Append(c))
+                    .ToString().ToLowerInvariant();
```

Comment:
I'm currently looking into why we're pulling the machine name in the linux scenario. I know it's b/c there's no WEBSITE_INSTANCE_ID in the test, but not sure what the expected behavior is.